plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/SourceRole.kt

```diff
@@ -12,6 +12,7 @@ import org.jetbrains.kotlin.formver.viper.ast.Info
 
 sealed interface SourceRole {
     data object ParamFunctionLeakageCheck : SourceRole
+    data class SubListCreation(val mayBeNegative: Boolean = false) : SourceRole
```

Comment:
Default variables like this typically make the code harder to read, especially with hard-to-intuit types like booleans. Could you use two source roles here?  You can have them be two implementations of one `SubListCreation` sealed interface.